ui/raidboss/data/triggers/o9s.js

```diff
@@ -182,16 +151,365 @@
+        if ( data.phaseType == 'enrage' ) {
+          return {
+            en: 'Spread (Everyone)',
+            de: 'Verteilen (Jeder)',
+            fr: 'Ecartez-vous (Tout le monde)',
+          };
+        }
+      },
+      run: function(data) {
+        if ( data.entropyCount > 1 )
+          delete data.entropyCount;
+        if ( data.phaseType == 'orb' || data.phaseType == 'enrage' )
+          delete data.entropyCount;
+      },
+    },
+    {
+      id: 'O9S entropy avoid hit',
+      regex: /:(\y{Name}) gains the effect of (?:Unknown_640|Entropy) from  for (\y{Float}) Seconds/,
+      regexDe: /:(\y{Name}) gains the effect of (?:Unknown_640|Chaosflammen) from  for (\y{Float}) Seconds/,
+      regexFr: /:(\y{Name}) gains the effect of (?:Unknown_640|Flammes du chaos) from  for (\y{Float}) Seconds/,
+      regexJa: /:(\y{Name}) gains the effect of (?:Unknown_640|混沌の炎) from  for (\y{Float}) Seconds/,
+      condition: function(data, matches) {
+        return matches[1] == data.me;
+      },
+      preRun: function(data) {
+        data.entropyCount = data.entropyCount || 0;
+      },
+      delaySeconds: function(data) {
+        if ( (data.role == 'tank' || data.role == 'healer') && data.phaseType == 'fire' )
+          return 19;
+        if ( (data.role != 'tank' && data.role != 'healer') && data.phaseType == 'fire' )
+          return 5;
+      },
+      infoText: function(data) {
+        if ( data.phaseType == 'fire' ) {
+          return {
+            en: 'Hide Middle',
+            de: 'Zur Mitte',
+            fr: 'Allez au centre',
+          };
+        }
+      },
+    },
+    {
+      id: 'O9S fire big bang',
+      regex: / 14:3180:Chaos starts using /,
+      regexJa: / 14:3180:カオス starts using /,
+      condition: function(data) {
+        return data.phaseType == 'fire';
+      },
+      alertText: function(data) {
+        return {
+          en: 'Hide Middle',
+          de: 'Zur Mitte',
+          fr: 'Allez au centre',
+        };
+      },
+    },
+
+    // Water Path
+    {
+      id: 'O9S water phase tracking',
+      regex: / 14:3187:Chaos starts using /,
+      regexJa: / 14:3187:カオス starts using /,
+      preRun: function(data) {
+        data.phaseCount = data.phaseCount || 0;
+        data.phaseCount += 1;
+        if ( data.phaseCount < 8 )
+          data.phaseType = 'water';
+      },
+    },
+    {
+      id: 'O9S dynamic fluid 1',
+      regex: /:(\y{Name}) gains the effect of (?:Unknown_641|Dynamic Fluid) from  for (\y{Float}) Seconds/,
+      regexDe: /:(\y{Name}) gains the effect of (?:Unknown_641|Chaosspritzer) from  for (\y{Float}) Seconds/,
+      regexFr: /:(\y{Name}) gains the effect of (?:Unknown_641|Eaux du chaos) from  for (\y{Float}) Seconds/,
+      regexJa: /:(\y{Name}) gains the effect of (?:Unknown_641|混沌の水) from  for (\y{Float}) Seconds/,
+      condition: function(data, matches) {
+        return matches[1] == data.me;
+      },
+      delaySeconds: function(data, matches) {
```

Comment:
Should there be a "stack donut" call for both the dps and the T/H ones, and not just the one that is on you?

Reply:
Yes, thats the whole point of having 2 dynamic fluid triggers. If there is the possibility to have 2 delays in one trigger I could fuse them.